tools/clang/lib/Parse/ParseDeclCXX.cpp

```diff
@@ -584,7 +584,13 @@ Decl *Parser::ParseUsingDeclaration(unsigned Context,
     }
 
     ConsumeToken();
-
+    // HLSL change begin -warning ext before HLSL2021.
+    if (getLangOpts().HLSL) {
```

Comment:
This looks like it's only under type alias declaration.  Does it cover the `using namespace n;` case?

Reply:
Fixed.